### ISSUE TITLE
Array not redirected correctly

### ISSUE THREAD

**Issue by cristianbuse** [WaynePhillipsEA/twinbasic#182]
**Describe the bug**
Array is not redirected correctly ```ByRef``` for the second time.

**To Reproduce**
```VBA
Module Test
    Sub Main()
        Dim arr() As Variant
        
        arr = Array(1, 2, 3)
        Debug.Print Join(arr, ", ") 'Prints: 1, 2, 3
        
        SwapValues arr(0), arr(2)
        Debug.Print Join(arr, ", ") 'Prints: 3, 2, 1
        
        Swap1DArrayValues arr, 0, 2
        Debug.Print Join(arr, ", ") 'Prints: 3, 2, 1. Should print: 1, 2, 3

    End Sub

    Public Sub SwapValues(ByRef val1 As Variant, ByRef val2 As Variant)
        Dim temp As Variant
        Dim needsSet1 As Boolean: needsSet1 = VBA.IsObject(val1)
        Dim needsSet2 As Boolean: needsSet2 = VBA.IsObject(val2)
        '
        If needsSet1 Then Set temp = val1 Else temp = val1
        If needsSet2 Then Set val1 = val2 Else val1 = val2
        If needsSet1 Then Set val2 = temp Else val2 = temp
    End Sub
    
    Public Sub Swap1DArrayValues(ByRef arr As Variant, ByVal index1 As Long, ByVal index2 As Long)
        If index1 <> index2 Then SwapValues arr(index1), arr(index2)
    End Sub
End Module
```

**Screenshots**
n/a

**Expected behavior**
The line ```Swap1DArrayValues arr, 0, 2``` swaps the first and third elements in VBA, but not in tB.

**Desktop (please complete the following information):**
 - OS: Windows 10 Enterprise V1909
 - twinBASIC compiler v0.10.871

**Additional context**
None

**Comment by WaynePhillipsEA** [comment]
Fixed in v0.10.933, thanks for reporting.

**Comment by cristianbuse** [comment]
The code above works fine in v0.10.933. But, I am afraid the following is also related but doesn't work:  
```VBA
Module Test
    Sub Main()
        Dim arr(0 to 1) As Variant
        
        arr(0) = 5
        Debug.Print TheTypeName(arr(0)) 'Works fine
        Debug.Print ElemTypeName(arr,0) 'Does not work
    End Sub
    
    Public Function ElemTypeName(ByRef arr As Variant, ByVal index As Long) As String
        Return TheTypeName(arr(index))
    End Function
    
    Public Function TheTypeName(ByRef v As Variant) As String
        If VBA.IsObject(v) Then
            Return "Object"
        Else
            Select Case VBA.VarType(v)
            Case vbNull
                Return "Null"
            Case vbEmpty
                Return "Empty"
            Case vbError
                Return "Error"
            Case vbBoolean
                Return "Bool"
            Case vbString
                Return "String"
            Case vbByte, vbInteger, vbLong, vbLongLong
                Return "Integer"
            Case vbCurrency, vbDecimal, vbDouble, vbSingle
                Return "Float"
            Case vbDate
                Return "Date"
            Case vbArray To vbArray + vbUserDefinedType
                Return "Array"
            Case vbUserDefinedType
                Return "UDT"
            Case vbDataObject
                Return "Custom Interface"
            Case Else
                Return "Impossible"
            End Select
        End If
    End Function
End Module
```

**Comment by WaynePhillipsEA** [comment]
Thanks - I'll take a look at that

**Comment by cristianbuse** [comment]
Just tested and the last code was working fine in v0.10.871. So, I'm inclined to say that a change in v0.10.933 is what is causing the issue.

**Comment by WaynePhillipsEA** [comment]
Fixed in v0.10.951, thanks for reporting

**Comment by cristianbuse** [comment]
Apologies - I must be annoying :smiley:. The same issue seems to be happening in a 3rd case (the 2 above work fine now):  
```VBA
Module Test
    Sub Main()
        Dim arr(0 to 0) As Variant
        arr(0) = "2"

        Debug.Print IsNumeric(arr(0))     'Prints: True
        Debug.Print IsNumeric2(arr(0))    'Prints: True
        Debug.Print IsElemNumeric(arr, 0) 'Prints: False - Wrong
    End Sub
    
    Public Function IsNumeric2(ByRef v As Variant) As Boolean
        Return IsNumeric(v)
    End Function
    
    Public Function IsElemNumeric(ByRef arr As Variant, ByVal index As Long) As Boolean
        Return IsNumeric(arr(index))
    End Function
End Module
```

**Comment by WaynePhillipsEA** [comment]
On the contrary!  Keep them coming :)

**Comment by WaynePhillipsEA** [comment]
Fixed in v0.10.959, thanks!
